app/ui/shared/widgets/search.jsx

```diff
@@ -29,7 +30,15 @@ const INPUT_STYLE = Style.registerStyle({
 
 class Search extends Component {
   shouldComponentUpdate(nextProps) {
-    return !isEqual(this.props, nextProps);
+    // Since this component uses plain js objects as props, we need to perform
+    // deep quality checks on them to make sure we don't unnecessarily rerender.
+    // However, there is no need to re-render this parent component when the
+    // children change, which would happen when doing naive deep equality.
+    // Furthermore, children in this component may contain immutable.js props,
+    // which are throwing warnings when accessing some of their properties
+    // while doing deep equality checks using lodash's `deepEqual`.
+    return !shallowEqual(this.props.children, nextProps.children) ||
+      !isEqual(omit(this.props, ['children']), omit(nextProps, ['children']));
```

Comment:
This is a non-intuitive sCU method, and since using it more than once, should probably go somewhere else sharable. Actually, a `lib/should-component-update` with some common patterns would be helpful, as we use the PureRenderMixin, simple isEqual, and then this more complicated scenario which could all be exported from the same place (and wouldn't need to remember about `fbjs` module path to shallowEqual)